processor/cumulativetodeltaprocessor/internal/tracking/identity.go

```diff
@@ -38,17 +40,22 @@ const A = int32('A')
 const SEP = byte(0x1E)
 const SEPSTR = string(SEP)
 
+func attributesIdentity(m pcommon.Map) string {
+	attrs := make([]string, 0, m.Len())
+	m.Range(func(k string, v pcommon.Value) bool {
+		attrs = append(attrs, k+":"+v.AsString())
+		return true
+	})
+	sort.Strings(attrs)
+	return strings.Join(attrs, SEPSTR)
+}
+
 func (mi *MetricIdentity) Write(b *bytes.Buffer) {
 	b.WriteRune(A + int32(mi.MetricType))
 	b.WriteByte(SEP)
 	b.WriteRune(A + int32(mi.MetricValueType))
-	mi.Resource.Attributes().Sort().Range(func(k string, v pcommon.Value) bool {
-		b.WriteByte(SEP)
-		b.WriteString(k)
-		b.WriteByte(':')
-		b.WriteString(v.AsString())
-		return true
-	})
```

Comment:
Curios why not sorting the attributes here? Since I believe this is a mutable component, and code is optimal that way since now we allocate lots of strings.

Reply:
Yes it mutates: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/cumulativetodeltaprocessor/factory.go#L34.  Sorting here definitely has a side effect that the component doesn't care about, but I also don't think it breaks anything.

Reply:
Yes, I was thinking about it as well. Added a note about this in the PR description. I think it's just not a clear side effect of a function that created an identifier. Maybe not the best for performance but makes the logic cleaner. I would rather keep the order of attributes as is but also can change it back to Sort if you both think otherwise

Reply:
I would rather stay with Sort here.

Reply:
Updated to keep Sort